proposals/0000-package-dependency-import-syntax.md

```diff
@@ -0,0 +1,159 @@
+# Inline package dependency import syntax
+
+* Proposal: [SE-NNNN](NNNN-filename.md)
+* Authors: [Rahul Malik](https://github.com/rahul-malik), [Ankit Aggarwal](https://github.com/aciidb0mb3r), [David Hart](https://github.com/hartbit)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+
+## Introduction
+
+Developers today are using Swift as a scripting language to utilize its language features and portability across macOS and Linux. Compared to other common scripting languages (Python, Ruby, etc.), Swift maintains a "batteries-not-included" approach and relies on external packages to provide functionality like command-line argument parsing that are common in scripts. This proposal focuses on a new syntax for importing package dependencies directly within scripts written in Swift with the same level of specificity offered by the current Package manifest schema.
+
+Swift-evolution thread: [Discussion thread topic for that
+proposal](https://forums.swift.org/)
+
+## Motivation
+
+Writing scripts in Swift currently requires more work than other languages. The community has filled many of the gaps in terms of functionality like command-line argument parsing but even utilizing these packages can add significantly more friction.
+
+The community has also tried to solve this problem themselves through projects like [Marathon](http://github.com/johnsundell/marathon) and [swift-sh](http://github.com/mxcl/swift-sh) to provide functionality like resolving and fetching dependencies which are available within SwiftPM today.
+There are drawbacks to this solution
+- Additional tooling is required for any host machine that runs a Swift script
+- Package resolution and fetching is performed outside of SwiftPM and can't leverage it's capabilities
```

Comment:
Minor tweak suggestions:

> The Swift community has also recognized and tried to provide better scripting support through projects like [Marathon](http://github.com/johnsundell/marathon) and [swift-sh](http://github.com/mxcl/swift-sh). These projects leverage the Swift Package Manager to fetch and build external packages referenced in a Swift script. We believe we can take this approach further and provide a much better scripting experience by improving the integration between the Swift language and the Swift Package Manager. This will also allow Swift users to easily share their scripts without requiring to install additional tooling.